src/plugins/data_source_management/public/components/utils.ts

```
@@ -123,11 +123,14 @@ export const fetchDataSourceConnections = async (
   hideLocalCluster: boolean
 ) => {
   try {
-    const directQueryConnectionsPromises = dataSources.map((ds) =>
-      getDirectQueryConnections(ds.id, http!).catch(() => [])
-    );
-    const directQueryConnectionsResult = await Promise.all(directQueryConnectionsPromises);
-    const directQueryConnections = directQueryConnectionsResult.flat();
+    const directQueryConnections: DataSourceTableItem[] = [];
+
+    for await (const datasource of dataSources) {
+      await getDirectQueryConnections(datasource.id, http!)
+        .then((connections) => directQueryConnections.push(...connections))
+        .catch(() => directQueryConnections.push([]));
```

Do we want to swallow exceptions here?